basex-core/src/test/java/org/basex/server/SessionTest.java

```diff
@@ -545,27 +539,25 @@ public void queryBind2() throws IOException {
     }
   }
 
-  /** Runs an erroneous query.
-   * @throws IOException expected exception*/
-  @Test(expected = BaseXException.class)
-  public void queryError() throws IOException {
-    session.query("(").next();
+  /** Runs an erroneous query. */
+  @Test
+  public void queryError() {
+    assertThrows(BaseXException.class, () -> session.query("(").next());
   }
 
-  /** Runs an erroneous query.
-   * @throws IOException expected exception */
-  @Test(expected = BaseXException.class)
-  public void queryError2() throws IOException {
-    session.query("(1,'a')[. eq 1]").execute();
+  /** Runs an erroneous query. */
+  @Test
+  public void queryError2() {
+    assertThrows(BaseXException.class, () -> session.query("(1,'a')[. eq 1]").execute());
   }
 
-  /** Runs an erroneous query.
-   * @throws IOException expected exception*/
-  @Test(expected = BaseXException.class)
+  /** Runs an erroneous query. */
+  @Test
+  // FIXME: it seems that an error is thrown on the first next() call even though it should be ok
```

Comment:
@dimitarp The test doesn’t trigger an error on my system. How can this be reproduced?

Reply:
…and I was too quick. I’ll have a look into it.

Reply:
Fixed: The existing test was deficient, as it was the first `next()` call that raised the error.

This was different in (much) earlier versions of BaseX, in which the query API didn’t cache all results before delivering them to the caller.

Reply:
yes, I guessed so, but I thought you could know better what is the cause ;)